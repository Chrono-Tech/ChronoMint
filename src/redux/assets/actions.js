import contractsManagerDAO from 'dao/ContractsManagerDAO'

export const ASSETS_LIST = 'assets/LIST'

export const listAssets = () => async (dispatch) => {
  const dao = await contractsManagerDAO.getAssetsManagerDAO()
  dispatch({type: ASSETS_LIST, list: await dao.getList()})
}
