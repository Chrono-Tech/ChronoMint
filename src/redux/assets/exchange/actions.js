import type ExchangeManagerDAO from 'dao/ExchangeManagerDAO'
import type ExchangeModel from 'models/ExchangeModel'
import type TokenModel from 'models/TokenModel'
import contractsManagerDAO from 'dao/ContractsManagerDAO'

export const EXCHANGE_MANAGER_LIST = 'exchangeManager/LIST'
export const EXCHANGE_MANAGER_SET = 'exchangeManager/SET'
export const EXCHANGE_MANAGER_REMOVE = 'exchangeManager/REMOVE'

const getDAO = async (): Promise<ExchangeManagerDAO> => await contractsManagerDAO.getExchangeManagerDAO()

export const listExchanges = () => async (dispatch) => {
  const dao = await getDAO()
  dispatch({type: EXCHANGE_MANAGER_LIST, list: await dao.getList()})
}

export const addExchange = (address) => async () => {
  const exchange = new ExchangeModel({address, isFetching: true})
  dispatch({type: EXCHANGE_MANAGER_SET, exchange})
  try {
    const dao = await getDAO()
    await dao.addExchange(address)
  } catch (e) {
    // no rollback
  }
}

export const createExchange = (token: TokenModel) => async () => {
  try {
    const dao = await getDAO()
    await dao.createExchange(token)
  } catch (e) {
    // no rollback
  }
}

export const editExchange = (exchange: ExchangeModel, newExchange: ExchangeModel) => async () => {
  try {
    const dao = await getDAO()
    await dao.editExchange(exchange, newExchange)
  } catch (e) {
    // no rollback
  }
}

export const removeExchange = (exchange: ExchangeModel) => async () => {
  try {
    const dao = await getDAO()
    await dao.removeExchange(exchange)
  } catch (e) {
    // no rollback
  }
}

// TODO @bshevchenko: MINT-230 AssetsManager
// export const addOwner = (exchange: ExchangeModel, account) => async () => {
//   try {
//     const dao = await getDAO()
//     await dao.addOwner(exchange, account)
//   } catch (e) {
//     // no rollback
//   }
// }
//
// export const removeOwner = (exchange: ExchangeModel, account) => async () => {
//   try {
//     const dao = await getDAO()
//     await dao.removeOwner(exchange, account)
//   } catch (e) {
//     // no rollback
//   }
// }
