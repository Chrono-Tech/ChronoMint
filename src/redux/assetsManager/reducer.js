import Immutable from 'immutable'

import {
  GET_ASSETS_MANAGER_COUNTS,
  GET_ASSETS_MANAGER_COUNTS_START,
  GET_MANAGERS_FOR_TOKEN,
  GET_MANAGERS_FOR_TOKEN_LOADING,
  GET_PLATFORMS,
  GET_TOKENS,
  GET_TRANSACTIONS_DONE,
  GET_TRANSACTIONS_START,
  GET_USER_PLATFORMS,
  SELECT_PLATFORM,
  SELECT_TOKEN,
  SET_FEE,
  SET_IS_REISSUABLE,
  SET_NEW_MANAGERS_LIST,
  SET_TOKEN,
  SET_TOTAL_SUPPLY,
  SET_WATCHERS,
} from './actions'

const initialState = {
  usersPlatforms: [],
  usersPlatformsCount: 0,
  assetsManagerCountsLoading: false,
  selectedToken: null,
  selectedPlatform: null,
  tokensCount: 0,
  managersCount: 0,
  tokensOnCrowdsaleCount: 0,
  platformsList: [],
  tokensMap: new Immutable.Map(),
  managersList: [],
  assets: {},
  managersForTokenLoading: false,
  watchers: {},
  transactionsList: [],
  transactionsFetched: false,
  transactionsFetching: false,
}

export default (state = initialState, action) => {
  switch (action.type) {
    case GET_ASSETS_MANAGER_COUNTS_START:
      return {
        ...initialState,
        ...state,
        assetsManagerCountsLoading: true,
      }
    case GET_ASSETS_MANAGER_COUNTS:
      return {
        ...initialState,
        ...state,
        assetsManagerCountsLoading: false,
        tokensCount: Object.keys(action.payload.assets).length,
        managersCount: action.payload.managers.length,
        assets: action.payload.assets,
        managersList: action.payload.managers,
        platformsList: action.payload.platforms,
      }
    case GET_PLATFORMS:
      return {
        ...initialState,
        ...state,
        platformsList: action.payload.platforms,
      }
    case GET_MANAGERS_FOR_TOKEN_LOADING :
      return {
        ...initialState,
        ...state,
        managersForTokenLoading: true,
      }
    case SELECT_TOKEN:
      return {
        ...initialState,
        ...state,
        selectedToken: action.payload.symbol,
      }
    case SELECT_PLATFORM:
      return {
        ...initialState,
        ...state,
        selectedPlatform: action.payload.platformAddress,
      }
    case GET_TOKENS:
      return {
        ...initialState,
        ...state,
        tokensCount: action.payload.tokensMap.size,
        tokensMap: action.payload.tokensMap,
        assets: action.payload.assets,
      }
    case SET_TOKEN:
      return {
        ...initialState,
        ...state,
        tokensCount: state.tokensMap.concat(action.payload.tokensMap).size,
        tokensMap: state.tokensMap.concat(action.payload.tokensMap),
        assets: action.payload.assets,
      }
    case GET_MANAGERS_FOR_TOKEN:
      return {
        ...initialState,
        ...state,
        tokensMap: state.tokensMap.setIn([action.payload.symbol, 'managersList'], action.payload.managersForAssetSymbol),
        managersForTokenLoading: false,
      }
    case SET_WATCHERS:
      return {
        ...initialState,
        ...state,
        watchers: {
          ...state.watchers,
          ...action.payload.watchers,
        },
      }
    case SET_TOTAL_SUPPLY:
      return {
        ...initialState,
        ...state,
        tokensMap: state.tokensMap.setIn([action.payload.symbol, 'totalSupply'], action.payload.totalSupply),
        assets: {
          ...state.assets,
          [state.tokensMap.get(action.payload.symbol).address()]: {
            ...state.assets[action.payload.symbol],
            totalSupply: action.payload.totalSupply,
          },
        },
      }
    case SET_IS_REISSUABLE:
      return {
        ...initialState,
        ...state,
        tokensMap: state.tokensMap.setIn([action.payload.symbol, 'isReissuable'], action.payload.isReissuable),
      }
    case SET_FEE:
      let newTokensMap = state.tokensMap.setIn([action.payload.symbol, 'fee'], action.payload.fee)
      newTokensMap = newTokensMap.setIn([action.payload.symbol, 'withFee'], action.payload.withFee)
      return {
        ...initialState,
        ...state,
        tokensMap: newTokensMap,
      }
    case SET_NEW_MANAGERS_LIST:
      return {
        ...initialState,
        ...state,
        tokensMap: state.tokensMap.setIn([action.payload.symbol, 'managersList'], action.payload.managersList),
        managersList: action.payload.managers,
        managersCount: action.payload.managers.length,
      }
    case GET_TRANSACTIONS_START:
      return {
        ...initialState,
        ...state,
        transactionsList: [],
        transactionsFetched: false,
        transactionsFetching: true,
      }
    case GET_TRANSACTIONS_DONE:
      return {
        ...initialState,
        ...state,
        transactionsList: [
          ...state.transactionsList,
          ...action.payload.transactionsList,
        ],
        transactionsFetched: true,
        transactionsFetching: false,
      }
    case GET_USER_PLATFORMS:
      return {
        ...initialState,
        ...state,
        usersPlatforms: action.payload.usersPlatforms,
        usersPlatformsCount: action.payload.usersPlatforms.length,
      }
    default:
      return state
  }
}
