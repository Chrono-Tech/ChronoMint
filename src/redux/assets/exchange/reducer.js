import Immutable from 'immutable'
import * as a from './actions'

const initialState = {
  list: new Immutable.Map(),
  isFetched: false
}

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case a.EXCHANGE_MANAGER_LIST:
      return {
        ...state,
        list: action.list,
        isFetched: true
      }
    default:
      return state
  }
}

export default reducer
