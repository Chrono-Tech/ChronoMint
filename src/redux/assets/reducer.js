import Immutable from 'immutable'
import * as a from './actions'

const initialState = {
  list: new Immutable.Map(),
  isFetched: false
}

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case a.ASSETS_LIST:
      return {
        ...state,
        list: action.list,
        isFetched: true
      }
    default:
      return state
  }
}

export default reducer
