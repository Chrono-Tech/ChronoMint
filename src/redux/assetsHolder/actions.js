import { DUCK_SESSION } from '@/redux/session/actions'
import contractsManagerDAO from 'dao/ContractsManagerDAO'
import TIMEHolderDAO from 'dao/TIMEHolderDAO'
import Amount from 'models/Amount'
import ApprovalNoticeModel from 'models/notices/ApprovalNoticeModel'
import TokenModel from 'models/tokens/TokenModel'
import { DUCK_TOKENS } from 'redux/tokens/actions'
import tokenService, { EVENT_NEW_TOKEN } from 'services/TokenService'

export const DUCK_ASSETS_HOLDER = 'assetsHolder'

export const TIME_HOLDER_INIT = 'timeHolder/INIT'
export const TIME_HOLDER_TIME_ADDRESS = 'timeHolder/timeAddress'
export const TIME_HOLDER_WALLET_ADDRESS = 'timeHolder/timeHolderWalletAddress'
export const TIME_HOLDER_ADDRESS = 'timeHolder/timeHolderAddress'

let timeHolderDAO: TIMEHolderDAO = null

const subscribeOnTokens = (token: TokenModel) => async (dispatch, getState) => {
  if (token.symbol() !== 'TIME') {
    return
  }

  dispatch({ type: TIME_HOLDER_TIME_ADDRESS, address: token.address() })

  // init
  const [ timeHolderAddress, timeHolderWalletAddress ] = await Promise.all([
    timeHolderDAO.getAddress(),
    timeHolderDAO.getWalletAddress(),
  ])

  dispatch({ type: TIME_HOLDER_ADDRESS, address: timeHolderAddress })
  dispatch({ type: TIME_HOLDER_WALLET_ADDRESS, address: timeHolderWalletAddress })

  // TODO @dkchv: review again
  const contractNames = {}
  contractNames[ timeHolderAddress ] = 'TIME Holder'
  ApprovalNoticeModel.setContractNames(contractNames)

  // update allowance
  const timeDAO = tokenService.getDAO(token.id())
  const { account } = getState().get(DUCK_SESSION)
  const [ timeHolderAllowance, timeHolderWalletAllowance ] = await Promise.all([
    timeDAO.getAccountAllowance(timeHolderAddress, account),
    timeDAO.getAccountAllowance(timeHolderWalletAddress, account),
  ])

  console.log('--assetHolder: ', +timeHolderAllowance, +timeHolderWalletAllowance)

  // token = token
  //   .setAllowance(timeHolderAddress, timeHolderAllowance)
  //   .setAllowance(timeHolderWalletAddress, timeHolderWalletAllowance)
}

export const initAssetsHolder = () => async (dispatch, getState) => {
  if (getState().get(DUCK_ASSETS_HOLDER).isInited()) {
    return
  }
  dispatch({ type: TIME_HOLDER_INIT, inInited: true })

  const { account } = getState().get(DUCK_SESSION)
  timeHolderDAO = await contractsManagerDAO.getTIMEHolderDAO()
  const [ assets, deposit ] = await Promise.all([
    timeHolderDAO.getSharesContract(),
    timeHolderDAO.getDeposit(account),
  ])
  console.log('--deposit', assets, +deposit)

  // subscribe to tokens
  const callback = (token) => dispatch(subscribeOnTokens(token))
  tokenService.on(EVENT_NEW_TOKEN, callback)
  // fetch for existing tokens
  const tokens = getState().get(DUCK_TOKENS)
  tokens.list().forEach(callback)
}

export const depositAsset = (amount: Amount, token: TokenModel) => async (dispatch) => {
  // const amountBN = new BigNumber(amount)

  // dispatch(balanceMinus(amountBN, token))
  const assetDAO = tokenService.getDAO(token)

  try {
    await timeHolderDAO.deposit(amount, assetDAO)
  } catch (e) {
    // eslint-disable-next-line
    console.error('deposit error', e.message)
  } finally {
    // compensation for update in watchTransfer
    // dispatch(balancePlus(amount, token))
  }
}

export const withdrawAsset = (amount: Amount, token: TokenModel) => async () => {
  // const amountBN = new BigNumber(amount)

  // dispatch(depositMinus(amountBN))
  const assetDAO = tokenService.getDAO(token)

  try {
    await timeHolderDAO.withdraw(amount, assetDAO)
  } catch (e) {
    // eslint-disable-next-line
    console.error('withdraw error', e.message)
  } finally {
    // dispatch(depositPlus(amountBN))
  }
}
