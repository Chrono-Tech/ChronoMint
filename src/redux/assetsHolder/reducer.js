import TimeHolderModel from 'models/timeHolder/TimeHolderModel'
import * as a from './actions'

const initialState = new TimeHolderModel()

export default (state = initialState, action) => {
  switch (action.type) {
    case a.TIME_HOLDER_INIT:
      return state.isInited(true)
    case a.TIME_HOLDER_TIME_ADDRESS:
      return state.timeAddress(action.address)
    case a.TIME_HOLDER_ADDRESS:
      return state.timeHolderAddress(action.address)
    case a.TIME_HOLDER_WALLET_ADDRESS:
      return state.timeHolderWalletAddress(action.address)
    default:
      return state
  }
}
