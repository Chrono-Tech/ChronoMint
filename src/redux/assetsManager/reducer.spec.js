import Immutable from 'immutable'
import TokenModel from 'models/tokens/TokenModel'
import * as a from './actions'
import reducer, { initialState } from './reducer'

describe('assetsManager reducer', () => {
  it('should return initial state', () => {
    expect(reducer(undefined, {})).toEqual(initialState)
  })

  it('should handle GET_ASSETS_MANAGER_COUNTS_START', () => {
    expect(
      reducer(null, {
        type: a.GET_ASSETS_MANAGER_COUNTS_START,
      }),
    ).toEqual({
      assetsManagerCountsLoading: true,
    })
  })

  it('should handle GET_ASSETS_MANAGER_COUNTS', () => {
    expect(
      reducer(null, {
        type: a.GET_ASSETS_MANAGER_COUNTS,
        payload: {
          assets: { a: '1' },
          managers: [ 'test', 'test1', 'test2' ],
          platforms: [ 'test', 'test1', 'test2' ],
        },
      }),
    ).toMatchSnapshot()
  })

  it('should handle GET_PLATFORMS', () => {
    expect(
      reducer(null, {
        type: a.GET_PLATFORMS,
        payload: {
          platforms: [ 'test', 'test1', 'test2' ],
        },
      }),
    ).toMatchSnapshot()
  })

  it('should handle GET_MANAGERS_FOR_TOKEN_LOADING', () => {
    expect(
      reducer(null, {
        type: a.GET_MANAGERS_FOR_TOKEN_LOADING,
      }),
    ).toEqual({
      managersForTokenLoading: true,
    })
  })

  it('should handle SELECT_TOKEN', () => {
    expect(
      reducer(null, {
        type: a.SELECT_TOKEN,
        payload: {
          symbol: 'LHT',
        },
      }),
    ).toEqual({
      selectedToken: 'LHT',
    })
  })

  it('should handle SELECT_PLATFORM', () => {
    expect(
      reducer(null, {
        type: a.SELECT_PLATFORM,
        payload: {
          platformAddress: 'test',
        },
      }),
    ).toEqual({
      selectedPlatform: 'test',
    })
  })

  it('should handle GET_TOKENS', () => {
    const map = new Immutable.Map({ a: 1, b: 2 })
    const result = reducer(null, {
      type: a.GET_TOKENS,
      payload: {
        tokensMap: map,
        assets: [ 1, 2, 3, 4 ],
      },
    })
    expect(result).toMatchSnapshot()
  })

  it('should handle SET_TOKEN', () => {
    const map = new Immutable.Map({ aaa: new TokenModel(), bbb: new TokenModel() })
    const mapAciont = new Immutable.Map({ ccc: new TokenModel() })
    const result = reducer(
      {
        tokensMap: map,
      },
      {
        type: a.SET_TOKEN,
        payload: {
          tokensMap: mapAciont,
          assets: [ 1, 2 ],
        },
      },
    )
    expect(result).toMatchSnapshot()
  })

  it('should handle GET_MANAGERS_FOR_TOKEN', () => {
    const map = new Immutable.Map({ LHT: new TokenModel({ managersList: [ 1 ] }) })
    const result = reducer(
      {
        tokensMap: map,
      },
      {
        type: a.GET_MANAGERS_FOR_TOKEN,
        payload: {
          symbol: 'LHT',
          managersForAssetSymbol: [ 1, 2, 3 ],
        },
      },
    )
    expect(result).toMatchSnapshot()
  })

  it('should handle SET_TOTAL_SUPPLY', () => {
    const token = new TokenModel({ symbol: 'LHT', address: '1234', totalSupply: 5 })
    const result = reducer(
      {
        tokensMap: new Immutable.Map({ LHT: token }),
        assets: {
          test: {
            totalSupply: 1,
          },
        },
      },
      {
        type: a.SET_TOTAL_SUPPLY,
        payload: {
          token: token,
        },
      },
    )
    expect(result).toMatchSnapshot()
  })

  it('should handle SET_IS_REISSUABLE', () => {
    const token = new TokenModel({ symbol: 'LHT' })
    const result = reducer(
      {
        tokensMap: new Immutable.Map({ LHT: token }),
      },
      {
        type: a.SET_IS_REISSUABLE,
        payload: {
          isReissuable: true,
          symbol: 'LHT',
        },
      },
    )
    expect(result).toMatchSnapshot()
  })

  it('should handle SET_FEE', () => {
    const token = new TokenModel({ symbol: 'LHT' })
    const result = reducer(
      {
        tokensMap: new Immutable.Map({ LHT: token }),
      },
      {
        type: a.SET_FEE,
        payload: {
          fee: 1,
          withFee: true,
          symbol: 'LHT',
        },
      },
    )
    expect(result).toMatchSnapshot()
  })

  it('should handle SET_NEW_MANAGERS_LIST', () => {
    const token = new TokenModel({ symbol: 'LHT' })
    const result = reducer(
      {
        tokensMap: new Immutable.Map({ LHT: token }),
      },
      {
        type: a.SET_NEW_MANAGERS_LIST,
        payload: {
          managers: [ 'test', 'test1', 'test2' ],
          managersList: [ 'test', 'test1', 'test2' ],
          symbol: 'LHT',
        },
      },
    )
    expect(result).toMatchSnapshot()
  })

  it('should handle GET_TRANSACTIONS_START', () => {
    expect(
      reducer(null, {
        type: a.GET_TRANSACTIONS_START,
      }),
    ).toEqual({
      transactionsFetching: true,
    })
  })

  it('should handle GET_TRANSACTIONS_DONE', () => {
    expect(
      reducer(
        { transactionsList: [] },
        {
          type: a.GET_TRANSACTIONS_DONE,
          payload: {
            transactionsList: [ { txHash: 'test' }, { txHash: 'test1' } ],
          },
        },
      ),
    ).toMatchSnapshot()
  })

  it('should handle GET_USER_PLATFORMS', () => {
    expect(
      reducer(null, {
        type: a.GET_USER_PLATFORMS,
        payload: {
          usersPlatforms: [ 'test', 'test1', 'test2' ],
        },
      }),
    ).toMatchSnapshot()
  })
})
