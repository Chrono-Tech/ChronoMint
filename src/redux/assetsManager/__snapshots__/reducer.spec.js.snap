// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`assetsManager reducer should handle GET_ASSETS_MANAGER_COUNTS 1`] = `
{
  "assets": {
    "a": "1"
  },
  "assetsManagerCountsLoading": false,
  "managersCount": 3,
  "managersList": [
    "test",
    "test1",
    "test2"
  ],
  "platformsList": [
    "test",
    "test1",
    "test2"
  ],
  "tokensCount": 1
}
`;

exports[`assetsManager reducer should handle GET_MANAGERS_FOR_TOKEN 1`] = `
{
  "managersForTokenLoading": false,
  "tokensMap": {
    "LHT": {
      "address": null,
      "allowance": {},
      "balance": "0",
      "blockchain": null,
      "dao": null,
      "decimals": null,
      "fee": null,
      "feeAddress": null,
      "feeRate": null,
      "icon": null,
      "id": null,
      "isFailed": false,
      "isFetched": false,
      "isFetching": false,
      "isOptional": true,
      "isPending": false,
      "isReissuable": false,
      "managersList": [
        1,
        2,
        3
      ],
      "name": null,
      "platform": null,
      "symbol": null,
      "timestamp": null,
      "totalSupply": "0",
      "transactionHash": null,
      "url": null,
      "withFee": false
    }
  }
}
`;

exports[`assetsManager reducer should handle GET_PLATFORMS 1`] = `
{
  "platformsList": [
    "test",
    "test1",
    "test2"
  ]
}
`;

exports[`assetsManager reducer should handle GET_TOKENS 1`] = `
{
  "assets": [
    1,
    2,
    3,
    4
  ],
  "tokensCount": 2,
  "tokensMap": {
    "a": 1,
    "b": 2
  }
}
`;

exports[`assetsManager reducer should handle GET_TRANSACTIONS_DONE 1`] = `
{
  "transactionsFetched": true,
  "transactionsFetching": false,
  "transactionsList": [
    {
      "txHash": "test"
    },
    {
      "txHash": "test1"
    }
  ]
}
`;

exports[`assetsManager reducer should handle GET_USER_PLATFORMS 1`] = `
{
  "usersPlatforms": [
    "test",
    "test1",
    "test2"
  ],
  "usersPlatformsCount": 3
}
`;

exports[`assetsManager reducer should handle SET_FEE 1`] = `
{
  "tokensMap": {
    "LHT": {
      "address": null,
      "allowance": {},
      "balance": "0",
      "blockchain": null,
      "dao": null,
      "decimals": null,
      "fee": 1,
      "feeAddress": null,
      "feeRate": null,
      "icon": null,
      "id": null,
      "isFailed": false,
      "isFetched": false,
      "isFetching": false,
      "isOptional": true,
      "isPending": false,
      "isReissuable": false,
      "managersList": null,
      "name": null,
      "platform": null,
      "symbol": "LHT",
      "timestamp": null,
      "totalSupply": "0",
      "transactionHash": null,
      "url": null,
      "withFee": true
    }
  }
}
`;

exports[`assetsManager reducer should handle SET_IS_REISSUABLE 1`] = `
{
  "tokensMap": {
    "LHT": {
      "address": null,
      "allowance": {},
      "balance": "0",
      "blockchain": null,
      "dao": null,
      "decimals": null,
      "fee": null,
      "feeAddress": null,
      "feeRate": null,
      "icon": null,
      "id": null,
      "isFailed": false,
      "isFetched": false,
      "isFetching": false,
      "isOptional": true,
      "isPending": false,
      "isReissuable": true,
      "managersList": null,
      "name": null,
      "platform": null,
      "symbol": "LHT",
      "timestamp": null,
      "totalSupply": "0",
      "transactionHash": null,
      "url": null,
      "withFee": false
    }
  }
}
`;

exports[`assetsManager reducer should handle SET_NEW_MANAGERS_LIST 1`] = `
{
  "managersCount": 3,
  "managersList": [
    "test",
    "test1",
    "test2"
  ],
  "tokensMap": {
    "LHT": {
      "address": null,
      "allowance": {},
      "balance": "0",
      "blockchain": null,
      "dao": null,
      "decimals": null,
      "fee": null,
      "feeAddress": null,
      "feeRate": null,
      "icon": null,
      "id": null,
      "isFailed": false,
      "isFetched": false,
      "isFetching": false,
      "isOptional": true,
      "isPending": false,
      "isReissuable": false,
      "managersList": [
        "test",
        "test1",
        "test2"
      ],
      "name": null,
      "platform": null,
      "symbol": "LHT",
      "timestamp": null,
      "totalSupply": "0",
      "transactionHash": null,
      "url": null,
      "withFee": false
    }
  }
}
`;

exports[`assetsManager reducer should handle SET_TOKEN 1`] = `
{
  "assets": [
    1,
    2
  ],
  "tokensCount": 3,
  "tokensMap": {
    "aaa": {
      "address": null,
      "allowance": {},
      "balance": "0",
      "blockchain": null,
      "dao": null,
      "decimals": null,
      "fee": null,
      "feeAddress": null,
      "feeRate": null,
      "icon": null,
      "id": null,
      "isFailed": false,
      "isFetched": false,
      "isFetching": false,
      "isOptional": true,
      "isPending": false,
      "isReissuable": false,
      "managersList": null,
      "name": null,
      "platform": null,
      "symbol": null,
      "timestamp": null,
      "totalSupply": "0",
      "transactionHash": null,
      "url": null,
      "withFee": false
    },
    "bbb": {
      "address": null,
      "allowance": {},
      "balance": "0",
      "blockchain": null,
      "dao": null,
      "decimals": null,
      "fee": null,
      "feeAddress": null,
      "feeRate": null,
      "icon": null,
      "id": null,
      "isFailed": false,
      "isFetched": false,
      "isFetching": false,
      "isOptional": true,
      "isPending": false,
      "isReissuable": false,
      "managersList": null,
      "name": null,
      "platform": null,
      "symbol": null,
      "timestamp": null,
      "totalSupply": "0",
      "transactionHash": null,
      "url": null,
      "withFee": false
    },
    "ccc": {
      "address": null,
      "allowance": {},
      "balance": "0",
      "blockchain": null,
      "dao": null,
      "decimals": null,
      "fee": null,
      "feeAddress": null,
      "feeRate": null,
      "icon": null,
      "id": null,
      "isFailed": false,
      "isFetched": false,
      "isFetching": false,
      "isOptional": true,
      "isPending": false,
      "isReissuable": false,
      "managersList": null,
      "name": null,
      "platform": null,
      "symbol": null,
      "timestamp": null,
      "totalSupply": "0",
      "transactionHash": null,
      "url": null,
      "withFee": false
    }
  }
}
`;

exports[`assetsManager reducer should handle SET_TOTAL_SUPPLY 1`] = `
{
  "assets": {
    "1234": {
      "totalSupply": 5
    },
    "test": {
      "totalSupply": 1
    }
  },
  "tokensMap": {
    "LHT": {
      "address": "1234",
      "allowance": {},
      "balance": "0",
      "blockchain": null,
      "dao": null,
      "decimals": null,
      "fee": null,
      "feeAddress": null,
      "feeRate": null,
      "icon": null,
      "id": null,
      "isFailed": false,
      "isFetched": false,
      "isFetching": false,
      "isOptional": true,
      "isPending": false,
      "isReissuable": false,
      "managersList": null,
      "name": null,
      "platform": null,
      "symbol": "LHT",
      "timestamp": null,
      "totalSupply": 5,
      "transactionHash": null,
      "url": null,
      "withFee": false
    }
  }
}
`;
